spec/fixtures/app/models/company.rb

```diff
@@ -8,4 +8,5 @@ class Company < ActiveRecord::Base
   has_many :users
 
   has_and_belongs_to_many :freelancers
+  has_and_belongs_to_many :worker_users, join_table: 'companies_users', association_foreign_key: 'user_id'
```

Comment:
Passing the  `join_table:` option here actually undermines the original intention of this PR, which was to make sure that the right join table is detected for Rails >= 4 without having this option set.
However, It should work now, even if it is not explicitly tested.